src/main/scala/mesosphere/marathon/MarathonModule.scala

```diff
@@ -340,42 +343,41 @@ class MarathonModule(conf: MarathonConf, http: HttpConf)
   @Singleton
   def provideTaskFailureRepository(
     @Named(ModuleNames.STORE_TASK_FAILURES) store: EntityStore[TaskFailure],
-    metrics: Metrics): TaskFailureRepository = {
-    new TaskFailureRepository(store, conf.zooKeeperMaxVersions.get, metrics)
+    metrics: Metrics): TaskFailureEntityRepository = {
+    new TaskFailureEntityRepository(store, conf.zooKeeperMaxVersions.get, metrics)
   }
 
   @Provides
   @Singleton
-  def provideAppRepository(
+  def provideAppEntityRepository(
```

Comment:
All of these injections are going away. Migrations will be the only consumer of the raw "XEntityRepository" - storageModule will only expose the regular repositories - and AppRepository will be read-only (GroupRepository is the only writer of AppRepository)